File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char,int>mpp;
        int odd=0;

        for(char ch:s){
            mpp[ch]++;
            if(mpp[ch]%2!=0){
                odd+=1;
            }
            else odd--;
        }

        return min(s.length(),s.length()-odd+1);
    }
};

File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character,Integer>freq=new HashMap<>();
        int n = s.length(), maxOddLength = 0, totalLength = 0;

        for (int i = 0; i < n; i++) {
            freq.put(s.charAt(i), freq.getOrDefault(s.charAt(i), 0) + 1);
        }

        boolean hasOdd=false;
        for(int value:freq.values()){
            if(value%2!=0){
                totalLength+=value-1;
                hasOdd=true;
            }
            else totalLength+=value;


        }

        return totalLength +((hasOdd) ? 1 : 0) ;
    }
}